testutils/clustermanager/gke.go

```diff
@@ -260,13 +261,70 @@ func (gc *GKECluster) Acquire() error {
 	return err
 }
 
-// Delete deletes a GKE cluster
-func (gc *GKECluster) Delete() error {
-	if !gc.NeedCleanup {
-		return nil
+// wait depends on unique opName(operation ID created by cloud), and waits until
+// it's done
+func (gc *GKECluster) wait(location, opName string, wait time.Duration) error {
+	validStatuses := []string{"PENDING", "RUNNING", "DONE"}
+	doneStatus := "DONE"
+	var op *container.Operation
+	var err error
+
+	timeout := time.After(wait)
+	tick := time.Tick(500 * time.Millisecond)
+	for {
+		select {
+		// Got a timeout! fail with a timeout error
+		case <-timeout:
+			return fmt.Errorf("timed out waiting")
+		case <-tick:
+			// Retry 3 times in case of weird network error, or rate limiting
```

Comment:
Wouldn't retries make rate limiting even more serious?

Reply:
That's why the interval doubles every cycle. I haven't ran into the rate limit yet, will make changes to this if it does